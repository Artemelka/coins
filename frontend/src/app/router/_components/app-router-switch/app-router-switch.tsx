import React, { FC, memo, ReactNode } from 'react';
import { Route, Switch, Redirect } from 'react-router';

type AppRouterSwitchProps = {
  children: ReactNode;
  notFoundPageComponent?: FC;
  pathToRedirect?: string;
};

export const AppRouterSwitchComponent: FC<AppRouterSwitchProps> = ({
  children,
  notFoundPageComponent,
  pathToRedirect,
}) => {
  return (
    <Switch>
      {children}
      {pathToRedirect && <Redirect to={pathToRedirect}/>}
      {notFoundPageComponent && (<Route component={notFoundPageComponent}/>)}
    </Switch>
  );
};

export const AppRouterSwitch = memo(AppRouterSwitchComponent);