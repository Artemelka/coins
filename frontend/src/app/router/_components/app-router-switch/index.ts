export { AppRouterSwitch } from './app-router-switch';