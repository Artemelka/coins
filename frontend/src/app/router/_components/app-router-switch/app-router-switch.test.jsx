import React from 'react';
import { Redirect } from 'react-router';
import { shallow } from 'enzyme';
import { AppRouterSwitchComponent } from './app-router-switch';

const getComponent = (props) => (
    <AppRouterSwitchComponent {...props}>
        <div>mock children</div>
    </AppRouterSwitchComponent>
);

const NotFoundPageComponent = () => <div>notFoundPageComponent</div>;

const pathToRedirect = 'test';

describe('AppRouterSwitchComponent', () => {
    describe('render', () => {
        test('empty', () => {
            const component = shallow(<AppRouterSwitchComponent/>);

            expect(component).toMatchSnapshot();
        });

        test('only children', () => {
            const component = shallow(getComponent());

            expect(component).toMatchSnapshot();
        });

        test('with redirect', () => {
            const component = shallow(getComponent({ pathToRedirect }));

            expect(component).toMatchSnapshot();
        });

        test('with notFoundPageComponent', () => {
            const component = shallow(getComponent({ notFoundPageComponent: NotFoundPageComponent }));

            expect(component).toMatchSnapshot();
        });

        test('with notFoundPageComponent & redirect', () => {
            const component = shallow(getComponent({
                notFoundPageComponent: NotFoundPageComponent,
                pathToRedirect,
            }));

            expect(component).toMatchSnapshot();
        });
    });

    test('prop "pathToRedirect" should be equal Redirect prop "to"', () => {
        const component = shallow(getComponent({ pathToRedirect }));

        expect(component.find(Redirect).props().to).toBe(pathToRedirect);
    });

});