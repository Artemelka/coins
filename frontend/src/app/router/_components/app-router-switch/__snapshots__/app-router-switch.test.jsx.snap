// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRouterSwitchComponent render empty 1`] = `<Switch />`;

exports[`AppRouterSwitchComponent render only children 1`] = `
<Switch>
  <div>
    mock children
  </div>
</Switch>
`;

exports[`AppRouterSwitchComponent render with notFoundPageComponent & redirect 1`] = `
<Switch>
  <div>
    mock children
  </div>
  <Redirect
    to="test"
  />
  <Route
    component={[Function]}
  />
</Switch>
`;

exports[`AppRouterSwitchComponent render with notFoundPageComponent 1`] = `
<Switch>
  <div>
    mock children
  </div>
  <Route
    component={[Function]}
  />
</Switch>
`;

exports[`AppRouterSwitchComponent render with redirect 1`] = `
<Switch>
  <div>
    mock children
  </div>
  <Redirect
    to="test"
  />
</Switch>
`;
